Source/WebKit/WebProcess/WebCoreSupport/WebRemoteFrameClient.cpp

```diff
@@ -95,6 +95,19 @@ String WebRemoteFrameClient::renderTreeAsText(size_t baseIndent, OptionSet<Rende
     return result;
 }
 
+String WebRemoteFrameClient::layerTreeAsText(size_t baseIndent, OptionSet<LayerTreeAsTextOptions> options)
+{
+    RefPtr page = m_frame->page();
+    if (!page)
+        return "Test Error - Missing page"_s;
+    options.add(LayerTreeAsTextOptions::IncludeRootLayers);
+    auto sendResult = page->sendSync(Messages::WebPageProxy::LayerTreeAsTextForTesting(m_frame->frameID(), baseIndent, options));
```

Comment:
I do not understand why do we have to have this complex workflow to dump the contents of the layers?

WebProcess: WebRemoteFrameClient::layerTreeAsText() -> sendMessage
UIProcess: WebPageProxy::layerTreeAsTextForTesting() -> sendMessage
WebProcess: WebPage::LayerTreeAsTextForTesting() -> completionHandler
UIProcess: WebPageProxy::layerTreeAsTextForTesting() ->completionHandler
WebProcess: WebRemoteFrameClient::layerTreeAsText()

Reply:
Is there another way?

The middle WebProcess in that list is a different process from the first/last, and we don't have direct WebProcess<->WebProcess IPC.

Reply:
Oh I see. May be in the furniture we can implement a new paradigm to handle WebProcess<->WebProcess IPC via UIProcess without having to repeat the same code you did for `layerTreeAsText`